src/codeu/chat/client/maingui/LoginPanel.java

```diff
@@ -0,0 +1,72 @@
+package codeu.chat.client.maingui;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+import javax.swing.*;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.common.LoginInputCallback;
+
+@SuppressWarnings("serial")
+public final class LoginPanel extends JPanel{
+
+    private final ClientContext clientContext;
+    private final LoginInputCallback loginInputCallback;
+
+    public LoginPanel(ClientContext clientContext, LoginInputCallback loginInputCallback) {
+        super();
+        this.clientContext = clientContext;
+        this.loginInputCallback = loginInputCallback;
+        initialize();
+    }
+    private void initialize() {
+
+        // This panel contains the login form for get access to the chat.
```

Comment:
Just a heads up, any kind of errors in the code should be corrected before merging. A grammar error in a comment is not a big deal but you don't ever want to merge "faulty" code to the upper branches.